Runtime/Scripts/Participant.cs

```diff
@@ -132,6 +139,108 @@ public void UpdateName(string name)
             var resp = request.Send();
         }
 
+        public PerformRpcInstruction PerformRpc(PerformRpcParams rpcParams)
+        {
+            using var request = FFIBridge.Instance.NewRequest<PerformRpcRequest>();
+            var rpcReq = request.request;
+            rpcReq.LocalParticipantHandle = (ulong)Handle.DangerousGetHandle();
+            rpcReq.DestinationIdentity = rpcParams.DestinationIdentity;
+            rpcReq.Method = rpcParams.Method;
+            rpcReq.Payload = rpcParams.Payload;
+            rpcReq.ResponseTimeoutMs = (uint)(rpcParams.ResponseTimeout * 1000);
+
+            using var response = request.Send();
+            FfiResponse res = response;
+            return new PerformRpcInstruction(res.PerformRpc.AsyncId);
+        }
+
+        public void RegisterRpcMethod(string method, RpcHandler handler)
+        {
+            _rpcHandlers[method] = handler;
+
+            using var request = FFIBridge.Instance.NewRequest<RegisterRpcMethodRequest>();
+            var registerReq = request.request;
+            registerReq.LocalParticipantHandle = (ulong)Handle.DangerousGetHandle();
+            registerReq.Method = method;
+            var resp = request.Send();
+        }
+
+        public void UnregisterRpcMethod(string method)
+        {
+            _rpcHandlers.Remove(method);
+
+            using var request = FFIBridge.Instance.NewRequest<UnregisterRpcMethodRequest>();
+            var unregisterReq = request.request;
+            unregisterReq.LocalParticipantHandle = (ulong)Handle.DangerousGetHandle();
+            unregisterReq.Method = method;
+            var resp = request.Send();
+        }
+
+        internal async void HandleRpcMethodInvocation(
```

Comment:
@cloudwebrtc @theomonnom  this is the piece I'm most unsure about. FFI triggers this, then we need to load the appropriate async user-supplied handler, execute it, then send its result back to FFI.  

I tried this with coroutines but as far as I could tell it required me to have a coroutine runner, which meant having a global singleton that might get overloaded from different calls? 

Instead for now it uses tasks with async/await but that makes it different than everything else

Is there a good way to have the user supply a coroutine that we can run without making a global singleton runner of our own? any other thoughts?